doc/developer/scan.txt

```diff
@@ -132,22 +63,266 @@ files, and section of the code they deal with, are :
 * ``scan_views.py`` contains different views of the scan op that have
   simpler and easier signatures to be used in specific cases.
 
-* ``scan_opt.py`` contains the list of all optimizations for the scan
-  operator.
+* ``scan_opt.py`` contains the list of all Theano graph optimizations for the
+  scan operator.
+
+
+Notation
+========
+
+Scan being a sizeable and complex module, it has its own naming convention for
+functions and variables which this section will attempt to introduce.
+
+Recall that a Scan op contains a Theano function representing the computation
+that is done in a single iteration of the loop represented by the Scan op (in
+other words, the computation given by the function provided as value to
+theano.scan's ''fn'' argument ). Whenever we discuss a scan op, "outer function"
+refers to the Theano function that *contains* the Scan op whereas the "inner
+function" refers to the Theano function that is *contained* in the Scan op.
+
+In the same spirit, the inputs to the Apply node wrapping the Scan op (or "Scan
```

Comment:
Maybe putting "Apply node wrapping the Scan op" in italic or some kind of emphasis to make clearer that "Scan node" refer to the whole thing and not just "Scan op"